cli/export.go

```diff
@@ -0,0 +1,431 @@
+package main
+
+import (
+	"archive/tar"
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"strconv"
+	"syscall"
+	"time"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/cheggaaa/pb"
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/docker/docker/pkg/term"
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/flynn/go-docopt"
+	"github.com/flynn/flynn/controller/client"
+	ct "github.com/flynn/flynn/controller/types"
+	"github.com/flynn/flynn/pkg/random"
+	"github.com/flynn/flynn/pkg/shutdown"
+)
+
+func init() {
+	register("export", runExport, `
+usage: flynn export [options]
+
+Export application configuration and data.
+
+The application's metadata, deploy strategy, release configuration, slug,
+formation, and Postgres database will be exported to a tar file.
+
+Options:
+	-f, --file=<file>  name of file to export to (defaults to stdout)
+	-q, --quiet        don't print progress
+`)
+
+	register("import", runImport, `
+usage: flynn import [options]
+
+Create a new application using exported configuration and data.
+
+The application will be created using the metadata, deploy strategy, release
+configuration, slug, formation, and Postgres database from the provided export
+file.
+
+Options:
+	-f, --file=<file>  name of file to import from (defaults to stdin)
+	-n, --name=<name>  name of app to create (defaults to exported app name)
+	-q, --quiet        don't print progress
+`)
+}
+
+func runExport(args *docopt.Args, client *controller.Client) error {
+	var dest io.Writer = os.Stdout
+	if filename := args.String["--file"]; filename != "" {
+		f, err := os.Create(filename)
```

Comment:
I don't think very many Unix-style tools do this. Also, it doesn't match the existing behavior of `flynn pg dump`.